vms/platformvm/state/state.go

```diff
@@ -2051,28 +2159,6 @@ func (s *state) writeCurrentStakers(updateValidators bool, height uint64) error
 				return err
 			}
 
-			if weightDiff.Amount == 0 {
-				// No weight change to record; go to next validator.
```

Comment:
Isn't this a behavior change? In the existing code, we return here if the weight didn't change. In the new code, we call one of `s.cfg.Validators.RemoveWeight`, `s.cfg.Validators.AddStaker` or `s.cfg.Validators.AddWeight`

Reply:
@danlaine,  you're right. I moved this condition to `writeWeightDiffs` but it should have been kept here too since it guards the cfg.Validators set update.
Rewritten the if/else as a switch as the linter suggest (and you do too next PR)